decidim-meetings/app/models/decidim/meetings/meeting.rb

```diff
@@ -153,9 +188,8 @@ def can_participate?(user)
         can_participate_in_space?(user) && can_participate_in_meeting?(user)
       end
 
-      def current_user_can_visit_meeting?(current_user)
-        (private_meeting? && registrations.exists?(decidim_user_id: current_user.try(:id))) ||
-          !private_meeting? || (private_meeting? && transparent?)
+      def current_user_can_visit_meeting?(user)
+        Decidim::Meetings::Meeting.visible_meeting_for(user).find_by(id: id)
```

Comment:
Probably more performant
```suggestion
        Decidim::Meetings::Meeting.visible_meeting_for(user).exists?(id: id)
```